webhook_handlers/views/github.py

```diff
@@ -411,6 +411,29 @@ def pull_request(self, request, *args, **kwargs):
 
         return Response()
 
+    def _decide_app_name(
+        self, ghapp: GithubAppInstallation, app_id: Union[str, int]
+    ) -> str:
+        """Possibly updated the name of a GithubAppInstallation that has been fetched from DB or created.
+        Only the real default installation maybe use the name `GITHUB_APP_INSTALLATION_DEFAULT_NAME`
+        (otherwise we break the app)
+        We check that apps:
+            * already were given a custom name (do nothing);
+            * app_id match the configured default app app_id (use default name);
+            * none of the above (use 'unconfigured_app');
+
+        Returns the app name that should be used
+        """
+        if ghapp.name != GITHUB_APP_INSTALLATION_DEFAULT_NAME:
+            return ghapp.name
+        installation_default_app_id = get_config("github", "integration", "id")
+        # `app_id` and `installation_default_app_id` should both be ints
+        # But just to avoid differences parsing either the YAML or the request data
+        # Casting them to str for the comparison (str has less change of failing)
```

Comment:
We probably want to check if pem_path is set here. If it is, we can also not change the name. Otherwise we are enforcing only one default app.